File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.length(), cnt = 0;
        for (int i=0;i<n;i++){
            unordered_set<int>us;
            for (int j=i;j<n;j++){
                us.insert(s[j]);
                if (us.size() == j-i+1){
                    cnt = max(cnt, j-i+1);
                }
                else {
                    break;
                }
            }
        }
        return cnt;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let mp = new Map();
    let i = 0, j = 0;
    var maxLength = 0;
    while (j < s.length){
        if (!mp.has(s[j])) mp.set(s[j],1);
        else {
            while (i < j && mp.has(s[j])){
                mp.delete(s[i])
                i++;
            }
            mp.set(s[j],1);
        }
        maxLength = Math.max(maxLength,mp.size);
        j++;
    }
    maxLength = Math.max(maxLength,mp.size);
    return maxLength;
};